### PR TITLE
Reintroduce full index compatible testing of full cluster restart

### PR DESCRIPTION
**Pull request by mark-vieira**
As a follow up to #82445 we are now adding back full cluster restart testing across all index-compatible versions of Elasticsearch but in a way compatible with https://github.com/elastic/elasticsearch/pull/82321. This means as part of doing a full cluster restart upgrade to 8.0 you must first upgrade to 7.last. All full cluster restart test scenarios have been updated to do exactly this.

### CODE REVIEW

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/full-bwc

File: plugins/examples/rescore/src/main/java/org/elasticsearch/example/rescore/ExampleRescoreBuilder.java [hunk 1]
```diff
@@ -173,10 +173,10 @@ public TopDocs rescore(TopDocs topDocs, IndexSearcher searcher, RescoreContext r
                             endDoc = leaf.docBase + leaf.reader().maxDoc();
                         } while (topDocs.scoreDocs[i].doc >= endDoc);
                         LeafFieldData fd = context.factorField.load(leaf);
-                        if (false == (fd instanceof LeafNumericFieldData leafNumericFieldData)) {
+                        if (false == (fd instanceof LeafNumericFieldData)) {
```
Comment (pugnascotia) [inline plugins/examples/rescore/src/main/java/org/elasticsearch/example/rescore/ExampleRescoreBuilder.java:176]: This looks like an unintentional revert? Or is it intentional?

**Review by grcevski — APPROVED** [review]
This looks good to me! LGTM!

File: plugins/examples/rescore/src/main/java/org/elasticsearch/example/rescore/ExampleRescoreBuilder.java (hunk 1 above)
Comment (mark-vieira) [reply plugins/examples/rescore/src/main/java/org/elasticsearch/example/rescore/ExampleRescoreBuilder.java:176]: Intentional, this change broke our example plugins so I just decided to fix it in this PR.

**Pull request merged**